app/locales/languages.js

```diff
@@ -31,44 +30,50 @@ import zh_Hant from './zh-Hant.json';
 // 5. REMOVE all empty translations. e.g. "key": "", this will allow fallback to the default: English
 // 6. import xyIndex from `./xy.json` and add the language to the block at the bottom
 
-const deviceLocale =
-  Platform.OS === 'ios'
-    ? NativeModules.SettingsManager.settings.AppleLocale || // iOS < 13
-      NativeModules.SettingsManager.settings.AppleLanguages[0] // iOS 13
-    : NativeModules.I18nManager.localeIdentifier; // Android
+// detect and set device locale to i18n and dates
+setLocale(getDeviceLocale());
 
-dayjs.locale([deviceLocale, 'en']);
+// detect user override and set i18n and date locales
+getUserLocaleOverride().then(locale => locale && setLocale(locale));
 
-// This will fetch the user's language
-// Set up as a function so first onboarding screen can also update
-// ...from async language override setting
-export function findUserLang(callback) {
-  let userLang = undefined;
-  getLanguages().then(languages => {
-    userLang = languages[0].split('-')[0]; // ['en-US' will become 'en']
+/** Fetch the user language override, if any */
+export async function getUserLocaleOverride() {
+  return await GetStoreData(LANG_OVERRIDE);
+}
 
-    // If the user specified a language override, use it instead
-    GetStoreData(LANG_OVERRIDE).then(res => {
-      if (typeof res === 'string') {
-        console.log('Found user language override:');
-        console.log(res);
-        userLang = res;
-        i18next.changeLanguage(res);
-        dayjs.locale(res);
-      } else {
-        i18next.changeLanguage(userLang);
-        dayjs.locale(userLang);
-      }
+/** Get the device locale (normalized, e.g. `en_US` -> `en`) */
+export function getDeviceLocale() {
+  /** full locale e.g. `en_US` */
+  const fullLocale =
+    Platform.OS === 'ios'
+      ? NativeModules.SettingsManager.settings.AppleLocale || // iOS < 13
+        NativeModules.SettingsManager.settings.AppleLanguages[0] // iOS 13
+      : NativeModules.I18nManager.localeIdentifier; // Android
+  return normalizeLocale(fullLocale);
+}
 
-      // Run state updating callback to trigger rerender
-      typeof callback === 'function' ? callback(userLang) : null;
+/**
+ * @param {string} fullLocale
+ */
+function normalizeLocale(fullLocale) {
+  const [first] = fullLocale.split('-');
```

Comment:
Is this a typo?  Shouldn't it be `fullLocale.split('_');`

Reply:
Good catch, I better double check this tomorrow

Reply:
I was supposed to be normalizing `en_US` to `en-us` for dayjs. Fixed.